### PR TITLE
Created an asset table and tidy up code

### PR DESCRIPTION
**Pull request by danielsteenberg**
Created a single table to store all assets and pull particular to view and neatened up code

### CODE REVIEW

File: app/models/asset.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class Asset < ActiveRecord::Base
+	belongs_to :employee
+	belongs_to :laptop
+end
```
Comment (jamieandyou) [inline app/models/asset.rb:3]: Shouldn't this model be deleted?
Looks like you need to do some cleaning up of all the references you have to Asset and Asset directory. 
And why did you decide to call it asset directory? Isn't just the word Asset good enough?
Chat to me if you want to clarify thought around this.

File: app/controllers/laptops_controller.rb [hunk 2]
```diff
@@ -1,30 +1,22 @@
...
 
-  # GET /laptops/1
-  # GET /laptops/1.json
   def show
   end
 
-  # GET /laptops/new
   def new
-    @laptop = Laptop.new
+    @laptop = AssetDirectory.new
```
Comment (jamieandyou) [inline app/controllers/laptops_controller.rb]: This might be confusing. I think we should check with Ryan if there is a better way to do this. Like maybe one controller that deals with Assets. Maybe you and I need to sit down and think about this as well. Let's make some time to sit together.

File: Gemfile [hunk 3]
```diff
@@ -30,6 +30,7 @@ gem 'nokogiri', '1.6.6.2'
 gem 'tzinfo-data', platforms: [:mingw, :mswin]
 
 group :development do
+	gem 'better_errors'
```
Comment (jamieandyou) [inline Gemfile]: Indentation

File: Gemfile [hunk 4]
```diff
@@ -30,6 +30,7 @@ gem 'nokogiri', '1.6.6.2'
...
```
Comment (jamieandyou) [inline Gemfile]: Remove this blank line

File: Gemfile.lock [hunk 5]
```diff
@@ -41,13 +41,18 @@ GEM
       execjs
     bcrypt (3.1.11)
     bcrypt (3.1.11-x86-mingw32)
+    better_errors (2.1.1)
```
Comment (jamieandyou) [inline Gemfile.lock:43]: Indentation of this whole file

File: app/controllers/assets_controller.rb [hunk 6]
```diff
@@ -0,0 +1,64 @@
+class AssetsController < ApplicationController
+  before_action :set_asset, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @assets = Asset.all
+  end
+
+  def laptops
+    @assets = Asset.where(category: 'Laptops')
```
Comment (jamieandyou) [inline app/controllers/assets_controller.rb:9]: Awesome :)

File: app/views/assets/_form.html.slim [hunk 7]
```diff
@@ -24,4 +24,5 @@
     .field
       = f.label :category
       = f.text_field :category
-    .actions = f.submit
+    .actions 
+    = f.submit
```
Comment (jamieandyou) [inline app/views/assets/_form.html.slim]: Add one indentation here so it falls within the .actions div

File: app/views/assets/laptops.html.slim [hunk 8]
```diff
@@ -0,0 +1,5 @@
+.container
```
Comment (jamieandyou) [inline app/views/assets/laptops.html.slim:1]: Indentation of this file is too large

File: app/views/assets/show.html.slim [hunk 9]
```diff
@@ -6,24 +6,24 @@
   .form
```
Comment (jamieandyou) [inline app/views/assets/show.html.slim:6]: Same. Indentation of this file is too large

File: app/views/assets/show.html.slim [hunk 10]
```diff
@@ -6,24 +6,24 @@
...
 		p
   		strong Item model:
```
Comment (jamieandyou) [inline app/views/assets/show.html.slim]: You need to indent the things under the p by one indentation so they fall within the p paragraph

File: app/views/shared/_asset_table.html.slim [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+  = link_to 'New Asset directory', new_asset_path, class: 'btn btn-primary'
+
+.float-right
+  h2 Search Bar
+  form.search-form
+    | Enter Search item here:
+    br
+    input name="username" type="text" 
+    br
+    input.button onclick="alert('Search')" type="button" value="Search"
```
Comment (jamieandyou) [inline app/views/shared/_asset_table.html.slim:38]: I just slacked you the user preferences you must put into your sublime.

The line: "ensure_newline_at_eof_on_save": true,
Will make sure that you don't get that little red arrow this at the end of a file when viewing on github.

So go back to this file after you've updated your user preferences and then save this file and it should create the new line on the bottom of the file automatically.

File: config/routes.rb [hunk 12]
```diff
@@ -14,9 +12,10 @@
 get "user/login"
 get "user/sign_up"
 get "employee/employee"
-get "laptop/laptop"
 get "employees/index"
 
+get 'assets/laptops', to: 'assets#laptops', as: 'laptops'
```
Comment (jamieandyou) [inline config/routes.rb]: I know this is going to change later to the search term parameters thing in the URL.

But for now I think it should be **/asset-directory/laptops** just because as a user you can see that it is another level deeper from /asset-directory

File: db/migrate/20160513081219_rename_old_table_to_new_table.rb [hunk 13]
```diff
@@ -0,0 +1,5 @@
+class RenameOldTableToNewTable < ActiveRecord::Migration
```
Comment (jamieandyou) [inline db/migrate/20160513081219_rename_old_table_to_new_table.rb:1]: The name for this file could be a lot better. For now it's fine. But in the future make it more descriptive.

File: db/schema.rb [hunk 14]
```diff
@@ -24,20 +24,9 @@
...
-    t.string   "item_description"
-    t.integer  "item_price"
-    t.string   "item_attributes"
-  end
-
   create_table "employees", force: :cascade do |t|
-    t.string   "first_name"
-    t.string   "last_name"
+    t.string   "fname"
+    t.string   "lname"
```
Comment (jamieandyou) [inline db/schema.rb]: Why did you change this?

fname and lname are not good. Should be first_name and last_name

File: Gemfile (hunk 3 above)
Comment (danielsteenberg) [inline Gemfile]: Done

File: Gemfile (hunk 4 above)
Comment (danielsteenberg) [inline Gemfile]: Done

File: Gemfile.lock (hunk 5 above)
Comment (danielsteenberg) [inline Gemfile.lock:43]: done

File: app/views/assets/_form.html.slim (hunk 7 above)
Comment (danielsteenberg) [inline app/views/assets/_form.html.slim]: Done

File: app/views/assets/laptops.html.slim (hunk 8 above)
Comment (danielsteenberg) [inline app/views/assets/laptops.html.slim:1]: Done

File: app/views/assets/show.html.slim (hunk 9 above)
Comment (danielsteenberg) [inline app/views/assets/show.html.slim:6]: Done

File: app/views/assets/show.html.slim (hunk 10 above)
Comment (danielsteenberg) [inline app/views/assets/show.html.slim]: On my side it is indented

File: app/views/shared/_asset_table.html.slim (hunk 11 above)
Comment (danielsteenberg) [inline app/views/shared/_asset_table.html.slim:38]: Coo, added it now

File: config/routes.rb (hunk 12 above)
Comment (danielsteenberg) [inline config/routes.rb]: it picks up under  /assets/laptops in the views now, because the folder is still called assets in the views

File: db/migrate/20160513081219_rename_old_table_to_new_table.rb (hunk 13 above)
Comment (danielsteenberg) [inline db/migrate/20160513081219_rename_old_table_to_new_table.rb:1]: Not a problem, next time i'll specify what table im changing

File: config/routes.rb (hunk 12 above)
Comment (danielsteenberg) [inline config/routes.rb]: routing error when changed to asset-directory

File: db/schema.rb (hunk 14 above)
Comment (danielsteenberg) [inline db/schema.rb]: Must have changed when i rolled back, i didnt change that. Changed it back now

File: Gemfile [hunk 15]
```diff
@@ -1,49 +1,31 @@
...
-
 gem 'devise'
-
 gem 'nokogiri', '1.6.6.2'
-
 gem 'tzinfo-data', platforms: [:mingw, :mswin]
 
 group :development do
- 
+	gem 'better_errors'
```
Comment (jamieandyou) [inline Gemfile]: indentation

File: app/models/asset.rb [hunk 16]
```diff
@@ -0,0 +1,3 @@
+class Asset < ActiveRecord::Base
+	belongs_to :employee
```
Comment (jamieandyou) [inline app/models/asset.rb]: indentation

File: app/models/employee.rb [hunk 17]
```diff
@@ -1,2 +1,3 @@
 class Employee < ActiveRecord::Base
+	has_many :assets
```
Comment (jamieandyou) [inline app/models/employee.rb]: indentation

File: app/views/assets/laptops.html.slim [hunk 18]
```diff
@@ -0,0 +1,5 @@
...
+	.page-header
```
Comment (jamieandyou) [inline app/views/assets/laptops.html.slim]: Still indentation problem

File: app/views/assets/show.html.slim [hunk 19]
```diff
@@ -0,0 +1,29 @@
...
+  		strong Assignee:
+  		= @asset.assignee
+		p
+  		strong Item Category:
+  		= @asset.category
+
+  br
+
+		= link_to 'Edit', edit_asset_path(@asset), class: 'btn btn-primary'
+		= link_to 'Done', laptops_path, class: 'btn btn-primary'
```
Comment (jamieandyou) [inline app/views/assets/show.html.slim]: still indentation problem

File: app/views/assets/show.html.slim [hunk 20]
```diff
@@ -0,0 +1,29 @@
...
+  		strong Item attributes:
+  		= @asset.item_attributes
+		p
+  		strong Assignee:
+  		= @asset.assignee
+		p
+  		strong Item Category:
+  		= @asset.category
+
+  br
```
Comment (jamieandyou) [inline app/views/assets/show.html.slim:25]: This br should be one level in from the .form div

File: config/routes.rb (hunk 12 above)
Comment (jamieandyou) [inline config/routes.rb]: Are you saying there is an error when you put the route as 
**get 'asset-directory/laptops', to: 'assets#laptops', as: 'laptops'**?

File: app/models/asset.rb (hunk 1 above)
Comment (danielsteenberg) [inline app/models/asset.rb:3]: Done

File: app/controllers/laptops_controller.rb (hunk 2 above)
Comment (danielsteenberg) [inline app/controllers/laptops_controller.rb]: Sorted

File: Gemfile.lock (hunk 5 above)
Comment (danielsteenberg) [inline Gemfile.lock:43]: The file gives an error after indentation, had to do bundle install and positioned back

File: app/views/assets/_form.html.slim [hunk 21]
```diff
@@ -1,4 +1,4 @@
-= form_for @asset, url: {action: "create"} do |f|
+= form_for @asset, :html => { :multipart => true , :class => 'form-horizontal' } do |f|
```
Comment (jamieandyou) [inline app/views/assets/_form.html.slim]: Rather don't use => if you don't have to 

So make it html: { multipart:        etc.

File: config/routes.rb [hunk 22]
```diff
@@ -1,76 +1,17 @@
 Rails.application.routes.draw do
 
-  scope "asset-directory" do
+  scope path: "asset-directory" do
     resources :assets, as: 'assets', path: 'all'
     get 'assets/laptops', to: 'assets#laptops', as: 'laptops', path: 'laptops'
```
Comment (jamieandyou) [inline config/routes.rb]: Why do you still have get assets/laptops?

Comment (danielsteenberg) [inline config/routes.rb]: Made some mistakes so undid all changes and started over and forgot to take that away

File: app/views/assets/_form.html.slim (hunk 21 above)
Comment (danielsteenberg) [inline app/views/assets/_form.html.slim]: sorted

File: app/views/assets/_form.html.slim [hunk 23]
```diff
@@ -1,4 +1,4 @@
-= form_for @asset, :html => { :multipart => true , :class => 'form-horizontal' } do |f|
+= form_for @asset, html: { :multipart => true , :class => 'form-horizontal' } do |f|
```
Comment (jamieandyou) [inline app/views/assets/_form.html.slim]: :multipart => true , :class => 'form-horizontal'

You can get rid of the => here too

File: app/views/assets/new.html.slim [hunk 24]
```diff
@@ -0,0 +1,5 @@
+h1 New asset
+
+== render 'form'
```
Comment (jamieandyou) [inline app/views/assets/new.html.slim]: Just use a single =

File: app/views/assets/edit.html.slim [hunk 25]
```diff
@@ -0,0 +1,8 @@
+h1 Editing asset
+
+== render 'form'
```
Comment (jamieandyou) [inline app/views/assets/edit.html.slim]: Just use a single =

File: spec/models/asset_directory_spec.rb [hunk 26]
```diff
@@ -0,0 +1,5 @@
+require 'spec_helper'
```
Comment (jamieandyou) [inline spec/models/asset_directory_spec.rb]: Delete all your spec files if you're not using them

File: db/migrate/20160420113806_create_asset_directories.rb [hunk 27]
```diff
@@ -0,0 +1,12 @@
+class CreateAssetDirectories < ActiveRecord::Migration
+  def change
+    create_table :asset_directories do |t|
```
Comment (jamieandyou) [inline db/migrate/20160420113806_create_asset_directories.rb:3]: Ideally you should be going back and making these three migrations into one migration, because all you did was make a table. But instead of having the one migration, you have a migration, then adding columns, then changing the table name, in one merge. You could be merging in just one migration to do essentially the same as the three.
But for the sake of progress, you can move on with it like this.

File: app/views/assets/_form.html.slim (hunk 23 above)
Comment (danielsteenberg) [inline app/views/assets/_form.html.slim]: Done

File: app/controllers/user_controller.rb [hunk 28]
```diff
@@ -1,7 +1,7 @@
 class UserController < ApplicationController
 
   def login
-	  # user.find_by(email: params[:session][:email].downcase)
+    # user.find_by(email: params[:session][:email].downcase)
```
Comment (jamieandyou) [inline app/controllers/user_controller.rb]: Get rid of this commented out code

**Comment by jamieandyou** [comment]
👍  
Side note: Your commit log will be quite long and bad, but it's the nature of all these changes. In the future you can learn how to squash all those commits into one nice commit with a good message.
